File: Zamba/ZambaWeb2013/zamba2010/Zamba.Core/SynchronizedHashtable.vb
Public Class SynchronizedHashtable
    'Inherits Hashtable

    Public _syncronizedHT As Hashtable

    Public Sub New()
        _syncronizedHT = Hashtable.Synchronized(New Hashtable)
    End Sub

    Public Sub Add(ByVal key As Object, ByVal value As Object)
        _syncronizedHT.Add(key, value)
    End Sub

    Default Public Property Item(ByVal key As Object) As Object
        Get
            Return _syncronizedHT.Item(key)
        End Get
        Set(ByVal value As Object)
            _syncronizedHT.Item(key) = value
        End Set
    End Property

    Public Sub Clear()
        _syncronizedHT.Clear()
    End Sub

    Public Function Contains(ByVal key As Object) As Boolean
        Return _syncronizedHT.Contains(key)
    End Function

    Public Function ContainsKey(ByVal key As Object) As Boolean
        Return _syncronizedHT.ContainsKey(key)
    End Function

    Public Function Clone() As Object
        Return _syncronizedHT.Clone()
    End Function

    Public Function ContainsValue(ByVal value As Object) As Boolean
        Return _syncronizedHT.ContainsValue(value)
    End Function

    Public ReadOnly Property Count As Integer
        Get
            Return _syncronizedHT.Count
        End Get
    End Property

    Public Sub Remove(ByVal key As Object)
        _syncronizedHT.Remove(key)
    End Sub

    Public Sub CopyTo(ByVal array As System.Array, ByVal arrayIndex As Integer)
        _syncronizedHT.CopyTo(array, arrayIndex)
    End Sub

    Public Function Equals(ByVal obj As Object) As Boolean
        Return _syncronizedHT.Equals(obj)
    End Function

    Public Function GetEnumerator() As System.Collections.IDictionaryEnumerator
        Return _syncronizedHT.GetEnumerator()
    End Function

    Public ReadOnly Property SyncRoot As Object
        Get
            Return _syncronizedHT.SyncRoot
        End Get
    End Property

    Public ReadOnly Property Values As ICollection
        Get
            Return _syncronizedHT.Values
        End Get
    End Property

    Public ReadOnly Property Keys As ICollection
        Get
            Return _syncronizedHT.Keys
        End Get
    End Property

End Class

File: Zamba/ZambaFramework/Zamba.FWEL/Zamba.FWEL/SynchronizedHashtable.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
public class SynchronizedHashtable
{
    //Inherits Hashtable


    public Hashtable _syncronizedHT;
    public SynchronizedHashtable()
    {
        _syncronizedHT = Hashtable.Synchronized(new Hashtable());
    }

    public void Add(object key, object value)
    {
        _syncronizedHT.Add(key, value);
    }

    public object this[object key]
    {
        get { return _syncronizedHT[key]; }
        set { _syncronizedHT[key] = value; }
    }

    public void Clear()
    {
        _syncronizedHT.Clear();
    }

    public bool Contains(object key)
    {
        return _syncronizedHT.Contains(key);
    }

    public bool ContainsKey(object key)
    {
        return _syncronizedHT.ContainsKey(key);
    }

    public object Clone()
    {
        return _syncronizedHT.Clone();
    }

    public bool ContainsValue(object value)
    {
        return _syncronizedHT.ContainsValue(value);
    }

    public int Count
    {
        get { return _syncronizedHT.Count; }
    }

    public void Remove(object key)
    {
        _syncronizedHT.Remove(key);
    }

    public void CopyTo(System.Array array, int arrayIndex)
    {
        _syncronizedHT.CopyTo(array, arrayIndex);
    }

    public bool Equals(object obj)
    {
        return _syncronizedHT.Equals(obj);
    }

    public System.Collections.IDictionaryEnumerator GetEnumerator()
    {
        return _syncronizedHT.GetEnumerator();
    }

    public object SyncRoot
    {
        get { return _syncronizedHT.SyncRoot; }
    }

    public ICollection Values
    {
        get { return _syncronizedHT.Values; }
    }

    public ICollection Keys
    {
        get { return _syncronizedHT.Keys; }
    }

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================

